### PR TITLE
Allow glue (e.g. A/AAAA) at NS record.

### PR DESCRIPTION
**Pull request by ximon18**

### CODE REVIEW

File: src/base/iana/rtype.rs [hunk 1]
```diff
@@ -425,3 +425,13 @@ int_enum! {
 }
 
 int_enum_str_with_prefix!(Rtype, "TYPE", b"TYPE", u16, "unknown record type");
+
+pub trait IsGlue {
```
Comment (partim) [inline src/base/iana/rtype.rs:429]: What’s the reason to have this as  a trait rather than just a normal method?

File: src/zonetree/parsed.rs [hunk 2]
```diff
@@ -123,36 +123,64 @@ impl Zonefile {
...
+                            .insert(record.owner().clone(), record.into());
+                        Ok(())
                     }
-                    self.cnames.insert(record.owner().clone(), record.into());
-                    Ok(())
                 }
                 _ => {
-                    if self.zone_cuts.contains(record.owner())
-                        || self.cnames.contains(record.owner())
+                    // Only gue records can only be added at the same owner as
```
Comment (partim) [inline src/zonetree/parsed.rs:165]: The comment probably broke during an edit …

File: src/base/iana/rtype.rs (hunk 1 above)
Comment (ximon18) [reply src/base/iana/rtype.rs:429]: Errm, none? I saw a macro and avoided it ;-)

Comment (ximon18) [reply src/base/iana/rtype.rs]: Done.

**Review by partim — APPROVED** [review]

**Pull request merged**